cli/cmd/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,72 @@
+package upgrade
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/consul-k8s/cli/cmd/common"
+	"github.com/hashicorp/go-hclog"
+)
+
+// TestValidateFlags tests the validate flags function.
+func TestValidateFlags(t *testing.T) {
+	// The following cases should all error, if they fail to this test fails.
+	testCases := []struct {
+		description string
+		input       []string
+	}{
+		{
+			"Should disallow non-flag arguments.",
+			[]string{"foo", "-auto-approve"},
+		},
+		{
+			"Should disallow specifying both values file AND presets.",
+			[]string{"-f='f.txt'", "-preset=demo"},
+		},
+		{
+			"Should error on invalid presets.",
+			[]string{"-preset=foo"},
+		},
+		{
+			"Should error on invalid timeout.",
+			[]string{"-timeout=invalid-timeout"},
+		},
+		{
+			"Should error on an invalid namespace.",
+			[]string{"-namespace=\" nsWithSpace\""},
```

Comment:
Can be removed if we remove namespace flag

Reply:
Removed